weld/pretty_print.rs

```diff
@@ -26,6 +26,12 @@ impl PrintableType for Type {
             Scalar(I64) => "i64".to_string(),
             Scalar(F32) => "f32".to_string(),
             Scalar(F64) => "f64".to_string(),
+            Vectorized(Bool) => "<? x bool>".to_string(),
+            Vectorized(I8) => "<? x i8>".to_string(),
+            Vectorized(I32) => "<? x i32>".to_string(),
+            Vectorized(I64) => "<? x i64>".to_string(),
+            Vectorized(F32) => "<? x f32>".to_string(),
+            Vectorized(F64) => "<? x f64>".to_string(),
```

Comment:
Do we want to allow parsing these too? I'd consider allowing that and choosing a simpler syntax, e.g. `simd[i32]`.

Also on a separate node you can factor this code to implement pretty_print for `ScalarKind` only once (perhaps just implement `Display` on it).